Python/Product/ExternalProfilerDriver/DWJsonModel.cs

```diff
@@ -0,0 +1,165 @@
+        public bool isUser { get; set; }
+        public IList<int> moduleIDs { get; set; }
+        public IList<ThreadSpec> threads { get; set; }
+    }
+
+    public class FrameInfo
+    {
+        public LongInt timestamp { get; set; }
+        public IList<LongInt> frameIPs { get; set; }
+    }
+
+    public class FunctionSpec
+    {
+        public string name { get; set; }
+        [JsonProperty("base")]
+        public LongInt baseX { get; set; }
+        public LongInt size { get; set; }
+
+        public FunctionSpec(string _name, long _base, long _size)
+        {
+            name = _name;
+            baseX = new LongInt(0, _base);
+            size = new LongInt(0, _size);
+        }
+    }
+
+    public class ModuleSpec
+    {
+        public string name { get; set; }
+        public int id { get; set; }
+        public LongInt begin { get; set; }
+        public LongInt end { get; set; }
+        [JsonProperty("base")]
+        public LongInt baseX { get; set; }
+        public LongInt size { get; set; }
+        public IList<FunctionSpec> ranges { get; set; }
+
+        public ModuleSpec()
+        {
+            /* empty */
+        }
+        public ModuleSpec(string _name, int _id, IEnumerable<FunctionSpec> _ranges)
+        {
+            name = _name;
+            id = _id;
+            ranges = _ranges.ToList();
+        }
+        public ModuleSpec(string _name, IEnumerable<FunctionSpec> _ranges)
+        {
+            name = _name;
+            ranges = _ranges.ToList();
+        }
+    }
+
+    public class ThreadSpec
+    {
+        public int id { get; set; }
+        public LongInt begin { get; set; }
+        public LongInt end { get; set; }
+        public IList<FrameInfo> stacks { get; set; }
+    }
+
+    public class Trace
+    {
+        public TimeSpec totalTimeRange { get; set; }
+        public string name { get; set; }
+        public ProcessorSpec processor { get; set; }
+        public IList<ProcessSpec> processes { get; set; }
+
+        public IList<ModuleSpec> modules { get; set; }
+    }
+
+    public class CPUUtilTrace
+    {
+        public LongInt beginTime;
+        public LongInt duration;
+        public IList<ValueTrace> counters;
+    }
+
+    public class ValueTrace
+    {
+        public string id;
+        public IList<CPUSample> p;
+        public ValueTrace(IEnumerable<CPUSample> samples)
+        {
+            id = "DiagnosticsHub.Counters.Process.CPU";
+            p = samples.ToList();
+        }
+    }
+
+    public class CPUSample
+    {
+        public LongInt t;
+        public float v;
+        public CPUSample(LongInt _t, float _v)
+        {
+            t = _t;
+            v = _v;
+        }
+    }
+
+    public static class TraceUtils
+    {
+        public static LongInt ToNanoseconds(double timepoint)
+        {
+            const ulong factor = 1000000000;
+            ulong timespec = (ulong)(timepoint * factor);
+            string hexrep = timespec.ToString("X");
```

Comment:
Check out the [`BitConverter`](https://docs.microsoft.com/en-us/dotnet/api/system.bitconverter?view=netframework-4.7) class for this conversion.